### PR TITLE
dev: fix `dev doctor` when run under a git worktree

### PR DESCRIPTION
**Pull request by irfansharif**
Broke it as part of 39fcab0dbb51e0a5a08b6dcad0fdf89f4dd99ca3.

Release note: None

### CODE REVIEW

**Review by rickystewart — APPROVED** [review]

**Review by andreimatei — COMMENTED** [review]
:lgtm: 

thanks!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/84755#-:-N7SG56Z6K_9PToWexVF:b-vkvjo7)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @rickystewart)

File: pkg/cmd/dev/doctor.go [hunk 1]
```diff
@@ -197,24 +197,30 @@ You can install node with: `+"`pkg install node`")
...
-		return err
-	}
-	hooks, err := d.os.ListFilesWithSuffix("githooks", "")
-	if err != nil {
-		return err
-	}
-	for _, hook := range hooks {
-		if err := d.os.Symlink(path.Join(workspace, hook), path.Join(gitHooksDir, path.Base(hook))); err != nil {
+	{
+		if _, err = d.exec.CommandContextSilent(ctx, "git", "rev-parse", "--is-inside-work-tree"); err != nil {
```
[the remark this answers is not in the document]
Comment (irfansharif) [reply pkg/cmd/dev/doctor.go:201]: It fails if we're not under a git repository. We were doing it previously too as part of the submodules check: https://github.com/cockroachdb/cockroach/commit/39fcab0dbb51e0a5a08b6dcad0fdf89f4dd99ca3#diff-e8039c332ff020d076e55ec8ea2ddd0c60be20ec64b6c04a2fe5df56c7a8061bL202.

**Pull request merged**